phoenix-core-client/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -780,6 +775,10 @@ public List<HRegionLocation> getTableRegions(byte[] tableName, byte[] startRowKe
             config.getInt(PHOENIX_GET_REGIONS_RETRIES, DEFAULT_PHOENIX_GET_REGIONS_RETRIES);
         TableName table = TableName.valueOf(tableName);
         byte[] currentKey = null;
+        final int queryTimeout = this.getProps().getInt(QueryServices.THREAD_TIMEOUT_MS_ATTRIB,
```

Comment:
querytimeout can be overridden at the phoenix statement level. We need to get it from the statementcontext `context.getStatement().getQueryTimeoutInMillis();`

Reply:
If it is not overridden at the statement level the statementcontext API  gets it from cqsi

Reply:
Sounds good, done